### PR TITLE
Impala schema2 import tools integration

### PR DESCRIPTION
**Pull request by svetlin-mladenov**
Schema2 is now fully supported by import tools which means studies can be imported using schema2.

This PR also contains a number of bug fixes.

### CODE REVIEW

**Review by lchorbadjiev — APPROVED** [review]
Let us merge this into the master. I have some minor remarks. You can address them before or after the merge.

File: dae/dae/import_tools/tests/test_task_graph.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+@pytest.fixture(params=["dask", "sequential"])
+def executor(threaded_client, request):
+    if request.param == "dask":
+        return DaskExecutor(threaded_client)
+    return SequentialExecutor()
+
+
+# NOTE these tests should not run in parallel
+EXEC_ORDER = []
```
[the remark this answers is not in the document]
Comment (svetlin-mladenov) [reply dae/dae/import_tools/tests/test_task_graph.py]: @lchorbadjiev Fixed

File: dae/dae/backends/schema2/serializers.py [hunk 2]
```diff
@@ -201,12 +204,11 @@ def schema_summary(self):
...
     @property
     def schema_family(self):
+        """Lazy construct and return the schema for the family alleles."""
         if self._schema_family is None:
             fields = []
-            for spr in self.FAMILY_SEARCHABLE_PROPERTIES_TYPES:
-                field = pa.field(
-                    spr, self.FAMILY_SEARCHABLE_PROPERTIES_TYPES[spr]
-                )
+            for spr, ftype in self.FAMILY_SEARCHABLE_PROPERTIES_TYPES.items():
```
[the remark this answers is not in the document]
Comment (svetlin-mladenov) [reply dae/dae/backends/schema2/serializers.py]: @lchorbadjiev Done

File: dae/dae/backends/schema2/serializers.py [hunk 3]
```diff
@@ -240,7 +242,9 @@ def _get_searchable_prop_value(self, allele, spr):
                 prop_value = prop_value.value
         return prop_value
 
-    def build_family_allele_batch_dict(self, allele, family_variant_data):
+    def build_family_allele_batch_dict(self, allele, family_variant_data) \
+            -> dict[str, list[Any]]:
+        """Build a batch of family allele data in the form of a dict."""
         family_header = []
         family_properties = []
```
[the remark this answers is not in the document]
Comment (svetlin-mladenov) [reply dae/dae/backends/schema2/serializers.py:213]: @lchorbadjiev I don't understand what you have in mind. `_get_searchable_prop_value` does a lot of stuff I don't see a clear way to remove it.

**Pull request merged**